### PR TITLE
backup: Rewrite erroneously expanded SQL stmts on-the-fly

### LINKED ISSUE #158: backup: cli tool failed to restore backup
## Setup:
- network `testnet`
- docker -  `alpine:3.11` (amd64)
- c-lighning [v0.9.1](https://github.com/ElementsProject/lightning/releases/tag/v0.9.1)
- plugins version [df4b422](https://github.com/lightningd/plugins/tree/df4b422)
- Python `3.8.2`
- pip freeze:
>Click==7.0
>flaky==3.7.0
>Mako==1.1.0
>MarkupSafe==1.1.1
>psutil==5.7.3
>pyln-client==0.8.2
>tqdm==4.51.0

## SQLite

- lib sqlite-dev [3.30.1](https://pkgs.alpinelinux.org/package/v3.11/main/armhf/sqlite) 
- python sqlite3.sqlite_version: `3.30.1`

## Steps to reproduce:
- backup-cli init performed
- backup plugin is working (db version >2000)
- lightningd is stopped 
- file `/var/lib/lightning/testnet/lightningd.sqlite3` is removed.

```bash
backup-cli restore file:///var/lib/lightning/backup/testnet.db /var/lib/lightning/testnet/lightningd.sqlite3
```

## Error:
The `lightningd.sqlite3` file is not restored, `backup-cli` crashed with error:
>unrecognized token: "1891733WHERE"

## Traceback:
```
662it [01:24,  7.85it/s]
Traceback (most recent call last):
  File "/usr/local/libexec/c-lightning/plugins/backup/backup-cli", line 76, in <module>
    cli()
  File "/usr/lib/python3.8/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3.8/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3.8/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3.8/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3.8/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/libexec/c-lightning/plugins/backup/backup-cli", line 64, in restore
    backend.restore(restore_destination)
  File "/usr/local/libexec/c-lightning/plugins/backup/backup.py", line 122, in restore
    self._restore_transaction(c.transaction)
  File "/usr/local/libexec/c-lightning/plugins/backup/backup.py", line 101, in _restore_transaction
    cur.execute(q.decode('UTF-8'))
sqlite3.OperationalError: unrecognized token: "1891733WHERE"
```

### PR DESCRIPTION
**Pull request by cdecker**
Rewrite queries that predate ElementsProject/lightning#4090 to include
the missing whitespace. Doing this on-the-fly seems to have little to
no impact on performance, so let's just do it this way. Has the
advantage of working transparently for the user.

Fixes #158

### CODE REVIEW

File: backup/test_backup.py [hunk 1]
```diff
@@ -207,3 +208,24 @@ def test_warning(directory, node_factory):
...
+            r'UPDATE outputs SET status=123, reserved_til=1891733WHERE prev_out_tx=1 AND prev_out_index=2',
+            r'UPDATE outputs SET status=123, reserved_til=1891733 WHERE prev_out_tx=1 AND prev_out_index=2',
+        ),
+    ]
+
+    b = FileBackend('destination', create=False)
+
+    for i, o in tests:
+        assert(b._rewrite_stmt(i) == o)
+
```
Comment (m-schmoock) [inline backup/test_backup.py:225]: Nit: E302 missing one more blank line :D

File: backup/test_backup.py [hunk 2]
```diff
@@ -207,3 +208,24 @@ def test_warning(directory, node_factory):
...
         r'The `--backup-destination` option is deprecated and will be removed in future versions of the backup plugin.'
     ))
+
+
+def test_rewrite():
+    tests = [
+        (
+            r'UPDATE outputs SET status=123, reserved_til=1891733WHERE prev_out_tx=1 AND prev_out_index=2',
+            r'UPDATE outputs SET status=123, reserved_til=1891733 WHERE prev_out_tx=1 AND prev_out_index=2',
+        ),
```
Comment (m-schmoock) [inline backup/test_backup.py:218]: Nit: you could also test for the second statement: `... peer_id=12345WHERE channels.id...`

**Review by m-schmoock — COMMENTED** [review]
ACK 5557bef
Kinda overkill for a bug that's already been fixed imho, but why not :D

**Comment by cdecker** [comment]
> ACK [5557bef](https://github.com/lightningd/plugins/commit/5557bef5a7eda7a692ef6478d6889488491824bf)
> Kinda overkill for a bug that's already been fixed imho, but why not :D

The issue is that users still have backups that predate the fix, so we need a way for them to restore those backups. Eventually we can retire this things, but for now it's required.

**Pull request merged**